app/ts-meta/meta/store.go

```diff
@@ -1035,6 +1035,25 @@ func (s *Store) leader() string {
 	return s.raft.Leader()
 }
 
+// leaderHTTP returns the http address what the Store thinks is the current leader. An empty
+// string indicates no leader exists.
+func (s *Store) leaderHTTP() string {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	if s.raft == nil {
+		return ""
+	}
+
+	leader := s.raft.Leader()
+	if leader == "" {
+		return leader
+	}
+
+	host := strings.Split(leader, ":")[0]
+	port := strings.Split(s.httpAddr, ":")[1]
+	return fmt.Sprintf("%s:%s", host, port)
```

Comment:
Because there may be multiple IPs, it is better to directly judge whether TCPHost equal to raft leader address in the data structure and take out the corresponding HttpHost.